lib/crutches/format/number.ex

```diff
@@ -418,4 +418,260 @@ defmodule Crutches.Format.Number do
+      "1,2 Million"
+
+      iex> Number.as_human(500000000, precision: 5)
+      "500 Million"
+
+      iex> Number.as_human(12345012345, significant: false)
+      "12.345 Billion"
+
+      iex> Number.as_human!("abc")
+      ** (ArithmeticError) bad argument in arithmetic expression
+  """
+  @as_human [
+    valid: [:locale, :precision, :significant, :separator, :delimiter,
+      :strip_insignificant_zeros, :units, :format],
+    defaults: [
+      precision: 3,
+      significant: true,
+      separator: ".",
+      delimiter: "",
+      strip_insignificant_zeros: true,
+      units: [
+        quadrillion: "Quadrillion",
+        trillion: "Trillion",
+        billion: "Billion",
+        million: "Million",
+        thousand: "Thousand",
+        hundred: "",
+        ten: "",
+        unit: "",
+        deci: "deci",
+        centi: "centi",
+        milli: "milli",
+        micro: "micro",
+        nano: "nano",
+        pico: "pico",
+        femto: "femto"
+      ],
+      format: "%n %u"
+    ]
+  ]
+
+  def as_human(number, opts \\ [])
+  def as_human(number, opts) when is_number(number) do
+    opts = Option.combine!(opts, @as_human)
+    {exp, unit, sign} = closest_size_and_sign(number)
+
+    precision = opts[:precision]
+
+    if precision < 0 do
+      precision = 0
+    end
+
+    delimited_opts = Keyword.take(opts, @as_delimited[:valid])
+
+    fract_num =
+      abs(number) / :math.pow(10, exp)
+      |> rounded_or_significant(opts[:significant], precision)
+      |> strip_trailing_zeros(opts[:strip_insignificant_zeros])
+      |> as_delimited(delimited_opts)
+
+    if sign < 0 do
+      fract_num = "-" <> fract_num
+    end
+
+    format_as_human(fract_num, opts[:units][unit], opts[:format])
+  end
+
+  defp closest_size_and_sign(number) when is_number(number) do
+    tenth_exp =
+      number
+      |> abs
+      |> :math.log10
+      |> trunc
+
+    sign = number_sign(number)
+
+    cond do
+      tenth_exp >= 15 -> {15, :quadrillion, sign}
+      tenth_exp >= 12 -> {12, :trillion, sign}
+      tenth_exp >= 9  -> {9, :billion, sign}
+      tenth_exp >= 6  -> {6, :million, sign}
+      tenth_exp >= 3  -> {3, :thousand, sign}
+      tenth_exp >= 2  -> {0, :hundred, sign}
+      tenth_exp >= 1  -> {0, :ten, sign}
+      tenth_exp >= 0  -> {0, :unit, sign}
+      tenth_exp < 0  && tenth_exp >= -1  -> {-1, :deci, sign}
+      tenth_exp < -1 && tenth_exp >= -2  -> {-2, :centi, sign}
+      tenth_exp < -2 && tenth_exp >= -3  -> {-3, :milli, sign}
+      tenth_exp < -3 && tenth_exp >= -6  -> {-6, :micro, sign}
+      tenth_exp < -6 && tenth_exp >= -9  -> {-9, :nano, sign}
+      tenth_exp < -9 && tenth_exp >= -12 -> {-12, :pico, sign}
+      tenth_exp < -12 -> {-15, :femto, sign}
+    end
+  end
+
+  defp number_sign(number) when is_number(number) do
+    cond do
+      number >= 0 ->  1
+      true -> -1
+    end
+  end
+
+  defp rounded_or_significant(number, significant, precision) do
```

Comment:
What do you think of function heads as a replacement for conditionals?

```elixir
defp rounded_or_significant(number, true, precision), do: make_significant(number, precision)
defp rounded_or_significant(number, false, precision) when precision <= 0 do
  number |> Float.round(precision) |> trunc |> Integer.to_string
end
defp rounded_or_significant(number, false, precision) do
  number |> Float.round(precision) |> :io_lib.format("~.#{precision}f", [number]) |> List.to_string
end
```

Also, now that I've implemented `Number.Format.as_rounded`, should this function use that?